app/controllers/work_packages/shares_controller.rb

```diff
@@ -92,4 +128,12 @@ def find_role_ids(builtin_value)
     # is why we unscope.
     WorkPackageRole.unscoped.where(builtin: builtin_value).pluck(:id)
   end
+
+  def find_shared_users
+    @shared_users = User
+                      .having_entity_membership(@work_package)
+                      .includes(work_package_shares: :roles)
+                      .where(work_package_shares: { entity: @work_package })
+                      .ordered_by_name
```

Comment:
For the use cases in here, where only the existence of any share is checked for, something like

```
Member.of_work_package(@work_package)
```
would suffice.

Given the uses cases, it might be even better to just memoize the count in this method.